plugin/pkg/admission/podplacementrestriction/admission.go

```diff
@@ -0,0 +1,258 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podplacementrestriction
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/controller/framework/informers"
+	"k8s.io/kubernetes/pkg/util/yaml"
+)
+
+func init() {
+	admission.RegisterPlugin("PodPlacementRestriction", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig := readConfig(config)
+		return NewPodTolerationsPlugin(client, pluginConfig.PodTolerationsPluginConfig["clusterDefaultTolerations"]), nil
+	})
+}
+
+var _ = admission.Interface(&podTolerationsPlugin{})
+var _ = admission.WantsInformerFactory(&podTolerationsPlugin{})
+
+type podTolerationsPlugin struct {
+	*admission.Handler
+	client            clientset.Interface
+	namespaceInformer framework.SharedIndexInformer
+	// global default tolerations in a cluster. If a namespace is
+	// not assigned tolerations, it gets this by default.
+	clusterDefaultTolerations []api.Toleration
+}
+
+type pluginConfig struct {
+	PodTolerationsPluginConfig map[string][]api.Toleration
+}
+
+// readConfig reads default value of clusterDefaultTolerations
+// from the file provided with --admission-control-config-file
+// If the file is not supplied, it defaults to "".
+// The format in a file:
+// podTolerationsPluginConfig:
+//  clusterDefaultTolerations:
+//    - Key: key1
+//      Value: value1
+//    - Key: key2
+//      Value: value2
+func readConfig(config io.Reader) *pluginConfig {
+	defaultConfig := &pluginConfig{}
+	if config == nil || reflect.ValueOf(config).IsNil() {
+		return defaultConfig
+	}
+	d := yaml.NewYAMLOrJSONDecoder(config, 4096)
+	for {
+		if err := d.Decode(defaultConfig); err != nil {
+			if err != io.EOF {
+				continue
```

Comment:
If I remember correctly, the reason I am doing this is because config file is common for all admission plugins, its difficult to differentiate if the error occurred due to this plugin config or some other, so ignoring them. That's one of the reason I preferred to have separate config file per plugin.